### PR TITLE
#312 Add "online" indicator; move Neuro SAN URL to Settings

### PR DESCRIPTION
**Pull request by dsargent**
# Overview
Adds a "status light" to the sidebar indicating status of connection to Neuro SAN. The status is checked every 30 seconds (configurable in the code). Settings for the Neuro SAN URL are also moved to the centralized Settings dialog, decluttering the sidebar.

# Demo

https://github.com/user-attachments/assets/bdf42d26-e1f5-4b10-a6dc-bdb187d550e4

# Screenshots

## Sidebar before/after
Before  | After
:-------------------------:|:-------------------------:
image|image

## Tooltip
image

## Error state
image

## Settings
image

# Notes

1) It doesn't prevent you from saving an invalid URL for Neuro SAN. This is intentional: it allows for edge cases where the user knows something we don't. For example, "the system is down right now so the test will fail, but I want to configure the URL for when it's back".

2) I changed the name of the persisted URL property and migrated it to our standard Zustand store approach, which changes the key name. I didn't feel it was worth providing a backward compatibility path for this (if someone happened to have saved a custom URL) given our tiny number of users and the even tinier subset that might have saved a custom URL.

3) Remove our custom `useLocalStorage` hook as there was only one place using it in the codebase (see (2) above) and instead, we should be using our standard solution, Zustand + auto persistence.

4) I thought of adding a "ping time" measure to the tooltip, but was already code complete. I may still do that.

### CODE REVIEW

File: packages/ui-common/components/MultiAgentAccelerator/MultiAgentAccelerator.tsx [hunk 1]
```diff
@@ -177,12 +176,7 @@ export const MultiAgentAccelerator: FC<MultiAgentAcceleratorProps> = ({
 
     const [providerKeysRequired, setProviderKeysRequired] = useState<ReadonlySet<LLMProvider>>(new Set())
 
-    const [customURLLocalStorage, setCustomURLLocalStorage] = useLocalStorage("customAgentNetworkURL", null)
-
-    // An extra set of quotes is making it in the string in local storage.
-    const [neuroSanURL, setNeuroSanURL] = useState<string>(
-        customURLLocalStorage?.replaceAll('"', "") || backendNeuroSanApiUrl
-    )
+    const neuroSanURL = useSettingsStore((state) => state.settings.externalServices.neuroSanUrl) ?? defaultNeuroSanUrl
```
[the remark this answers is not in the document]
Comment (dsargent) [reply packages/ui-common/components/MultiAgentAccelerator/MultiAgentAccelerator.tsx:179]: Fine, given our tiny user base.

File: __tests__/common/strictMocks.ts [hunk 2]
```diff
@@ -39,6 +39,8 @@ export const withStrictMocks = (options: StrictMockOptions = {}) => {
             jest.resetModules()
         }
 
+        localStorage.clear()
```
Comment (dsargent) [inline __tests__/common/strictMocks.ts:42]: Help keep tests pure

File: apps/main/__tests__/pages/MultiAgentAcceleratorPage.test.tsx [hunk 3]
```diff
@@ -63,7 +63,7 @@ describe("Multi Agent Accelerator Page", () => {
         expect(mockMultiAgentAcceleratorSpy).toHaveBeenCalledWith(
             expect.objectContaining<MultiAgentAcceleratorProps>({
                 username: MOCK_USER,
-                backendNeuroSanApiUrl: NEURO_SAN_SERVER_URL,
+                defaultNeuroSanUrl: NEURO_SAN_SERVER_URL,
```
Comment (dsargent) [inline apps/main/__tests__/pages/MultiAgentAcceleratorPage.test.tsx:66]: Renamed to something more meaningful

File: apps/main/pages/multiAgentAccelerator/index.tsx [hunk 4]
```diff
@@ -32,7 +32,7 @@ export const MultiAgentAcceleratorPage: FC & CustomPageProps = Object.assign(
         return (
             <MultiAgentAccelerator
                 username={username}
-                backendNeuroSanApiUrl={backendNeuroSanApiUrl}
+                defaultNeuroSanUrl={backendNeuroSanApiUrl}
```
Comment (dsargent) [inline apps/main/pages/multiAgentAccelerator/index.tsx:35]: Renamed

File: packages/ui-common/components/AgentChat/ChatCommon/ChatCommon.tsx [hunk 5]
```diff
@@ -860,6 +860,7 @@ export const ChatCommon = ({ref, ...props}: ChatCommonProps & {ref?: Ref<ChatCom
 
     const getOptionsMenuButton = () => (
         <Box
+            id={`${id}-options-menu-button-container`}
```
Comment (dsargent) [inline packages/ui-common/components/AgentChat/ChatCommon/ChatCommon.tsx:863]: Add target for Tour

File: packages/ui-common/components/Common/StatusLight.tsx
Comment (dsargent) [inline packages/ui-common/components/Common/StatusLight.tsx:1]: Extracted this to a common component, and now it's used in several places.

File: packages/ui-common/components/MultiAgentAccelerator/Tour/MainTourSteps.ts
Comment (dsargent) [inline packages/ui-common/components/MultiAgentAccelerator/Tour/MainTourSteps.ts:1]: Update the tour for the recent UI changes. A couple of the steps were silently broken, due to blind spots in the unit test (see `MultiAgentAcclerator.test.tsx`).

File: packages/ui-common/components/MultiAgentAccelerator/MultiAgentAccelerator.tsx [hunk 6]
```diff
@@ -69,7 +68,7 @@ import {MUIDialog} from "../Common/MUIDialog"
 
 export interface MultiAgentAcceleratorProps {
     readonly username: string
-    readonly backendNeuroSanApiUrl: string
+    readonly defaultNeuroSanUrl: string
```
Comment (dsargent) [inline packages/ui-common/components/MultiAgentAccelerator/MultiAgentAccelerator.tsx:71]: Renamed

File: packages/ui-common/components/MultiAgentAccelerator/MultiAgentAccelerator.tsx [hunk 7]
```diff
@@ -177,12 +176,7 @@ export const MultiAgentAccelerator: FC<MultiAgentAcceleratorProps> = ({
 
     const [providerKeysRequired, setProviderKeysRequired] = useState<ReadonlySet<LLMProvider>>(new Set())
 
-    const [customURLLocalStorage, setCustomURLLocalStorage] = useLocalStorage("customAgentNetworkURL", null)
```
Comment (dsargent) [inline packages/ui-common/components/MultiAgentAccelerator/MultiAgentAccelerator.tsx:180]: This stuff goes away and is now handled by the Settings dialog, which ends up putting the value in local storage for us.

File: packages/ui-common/components/Settings/ApiKeyInput.tsx [hunk 8]
```diff
@@ -68,6 +67,11 @@ export const ApiKeyInput: FC<ApiKeyInputProps> = ({
         setKeyValidated(null)
     }, [persistedValue])
 
+    const handleClearInput = () => {
```
Comment (dsargent) [inline packages/ui-common/components/Settings/ApiKeyInput.tsx:70]: Extract handler to function + fix bug: we were not resetting the "validated" status on clear input.

File: packages/ui-common/components/Settings/ApiKeyInput.tsx [hunk 9]
```diff
@@ -162,19 +166,10 @@ export const ApiKeyInput: FC<ApiKeyInputProps> = ({
                 value={inputValue}
                 variant="outlined"
             />
-            <Box
```
Comment (dsargent) [inline packages/ui-common/components/Settings/ApiKeyInput.tsx:165]: Extracted to new common component `StatusLight`.

File: packages/ui-common/components/Settings/SettingsDialog.tsx
Comment (dsargent) [inline packages/ui-common/components/Settings/SettingsDialog.tsx:1]: Add new section for Neuro SAN URL in here.

File: packages/ui-common/components/Settings/SettingsDialog.tsx [hunk 10]
```diff
@@ -101,7 +121,29 @@ const SettingsSubsection: FC<SettingsSubsectionProps> = ({title, children}) => (
         <SubSectionBody>{children}</SubSectionBody>
     </SubSection>
 )
+const URL_PROTOCOL_REGEX = /^https?:\/\//iu
```
Comment (dsargent) [inline packages/ui-common/components/Settings/SettingsDialog.tsx:124]: Utility stuff for handling the various URL options.

File: packages/ui-common/components/Settings/SettingsDialog.tsx [hunk 11]
```diff
@@ -151,6 +193,30 @@ export const SettingsDialog: FC<SettingsDialogProps> = ({id, isOpen, logoService
     const useNativeNames = useSettingsStore((state) => state.settings.appearance.useNativeNames)
     const nativeNamesCheckmark = useCheckmarkFade()
 
+    /*
```
Comment (dsargent) [inline packages/ui-common/components/Settings/SettingsDialog.tsx:196]: It is a little complicated due to these layered sources. This is essentially the logic that was in `Sidebar`, now resides here in `SettingsDialog` which seems more logical to me anyway.

File: packages/ui-common/components/Settings/SettingsDialog.tsx [hunk 12]
```diff
@@ -273,6 +339,54 @@ export const SettingsDialog: FC<SettingsDialogProps> = ({id, isOpen, logoService
         checkmark.trigger()
     }
 
+    const handleTestConnection = async () => {
```
Comment (dsargent) [inline packages/ui-common/components/Settings/SettingsDialog.tsx:347]: Handlers for new URL input section.

File: packages/ui-common/components/Settings/SettingsDialog.tsx [hunk 13]
```diff
@@ -273,6 +339,54 @@ export const SettingsDialog: FC<SettingsDialogProps> = ({id, isOpen, logoService
...
+        setNeuroSanUrlValidated(result.success)
+    }
+
+    const handleNeuroSanUrlChange = (e: ReactChangeEvent<HTMLInputElement>) => {
+        setNeuroSanUrlInput(e.target.value)
+        setNeuroSanUrlValidated(null)
+    }
+
+    const handleSaveNeuroSanUrl = () => {
+        const normalized = normalizeNeuroSanUrlInput(neuroSanUrlInput, neuroSanProtocol)
```
Comment (dsargent) [inline packages/ui-common/components/Settings/SettingsDialog.tsx:363]: One user comfort feature I added which wasn't present in the old dialog: if the user enters, or pastes, a full URL complete with protocol like `http://neuro-san.example.com` we "forgive" them (rather than issuing an error), strip the protocol and keep the bare URL, then set the dropdown to whatever protocol they entered. That is what `normalizeNeuroSanUrlInput()` is for.

We "normalize" (strip the protocol, if present) in `onBlur`, `onSave`, and when the user clicks "default".

File: packages/ui-common/components/Settings/SettingsDialog.tsx [hunk 14]
```diff
@@ -320,6 +434,97 @@ export const SettingsDialog: FC<SettingsDialogProps> = ({id, isOpen, logoService
         />
     )
 
+    const getNeuroSanSubsection = () => {
```
Comment (dsargent) [inline packages/ui-common/components/Settings/SettingsDialog.tsx:442]: New section for the Neuro SAN URL.

Note how easy and modular this is, thanks to the refactorings I did in the previous PR, so that we have reusable styles and components for sections. They are all used here for this new one.

File: packages/ui-common/components/Settings/SettingsDialog.tsx [hunk 15]
```diff
@@ -790,20 +995,23 @@ export const SettingsDialog: FC<SettingsDialogProps> = ({id, isOpen, logoService
         </SubSection>
     )
 
-    return (
-        // Always use default theme for settings dialog so user can always see to reset. It's possible that with
-        // certain custom themes the dialog would be unreadable.
-        <ThemeProvider
-            theme={createTheme({
+    const settingsTheme = useMemo(
```
Comment (dsargent) [inline packages/ui-common/components/Settings/SettingsDialog.tsx:1000]: Memoize this theme for performance; creating themes can be expensive and I may have been imagining it, but the dialog felt laggy before when entering text input.

File: packages/ui-common/components/Settings/SettingsRow.tsx [hunk 16]
```diff
@@ -32,6 +34,7 @@ export const SettingsRow: FC<SettingsRowProps> = ({
 
     return (
         <Box
+            data-testid={id}
```
Comment (dsargent) [inline packages/ui-common/components/Settings/SettingsRow.tsx:37]: Add ID for targeting in tests.

File: packages/ui-common/controller/agent/Agent.ts [hunk 17]
```diff
@@ -62,15 +62,16 @@ const insertTargetAgent = (agent: string, path: string) => {
...
     readonly status?: string
+    readonly success: boolean
     readonly version?: string
 }
 
 /**
  * Test connection for a neuro-san server.
  * @param url The neuro-san server URL.
- * @returns A boolean indicating whether the connection was successful.
+ * @returns An object with information about the connection test result
```
Comment (dsargent) [inline packages/ui-common/controller/agent/Agent.ts:74]: Doc was out of date

File: packages/ui-common/controller/agent/Agent.ts [hunk 18]
```diff
@@ -79,8 +80,13 @@ export const testConnection = async (url: string): Promise<TestConnectionResult>
     try {
         const response = await fetch(url, {signal: controller.signal})
         if (!response.ok) {
-            return {success: false, status: response.statusText}
+            return {
```
Comment (dsargent) [inline packages/ui-common/controller/agent/Agent.ts:83]: Now returns the HTTP status code as well, so we can show it to the user in the Tooltip status info. Helps with troubleshooting: there's a big difference between a 404 and a 503.

File: packages/ui-common/state/Settings.ts
Comment (dsargent) [inline packages/ui-common/state/Settings.ts:1]: Add new persisted item for the Neuro SAN URL (previously was using the `useLocalStorage` hook, hand-coded.)

File: packages/ui-common/utils/useLocalStorage.ts
Comment (dsargent) [inline packages/ui-common/utils/useLocalStorage.ts:1]: No longer needed 🎉 We use Zustand for our persistence.

File: packages/ui-common/index.ts [hunk 19]
```diff
@@ -54,5 +54,4 @@ export * from "./utils/Authentication"
 export * from "./utils/File"
 export * from "./utils/text"
 export * from "./utils/title"
-export * from "./utils/useLocalStorage"
```
Comment (dsargent) [inline packages/ui-common/index.ts:57]: Removed from the code and from the exports here.

File: jest.config.ts [hunk 20]
```diff
@@ -95,10 +95,10 @@ const config: Config.InitialOptions = {
 
     coverageThreshold: {
         global: {
-            statements: -115,
-            branches: -154,
-            functions: -29,
-            lines: -88,
+            statements: -114,
```
Comment (dsargent) [inline jest.config.ts:98]: Incremental gains.

File: packages/ui-common/__tests__/unit/Utils/UseLocalStorage.test.ts
Comment (dsargent) [inline packages/ui-common/__tests__/unit/Utils/UseLocalStorage.test.ts:1]: Module deleted, so delete the tests as well.

File: packages/ui-common/components/MultiAgentAccelerator/Sidebar/Sidebar.tsx
Comment (dsargent) [inline packages/ui-common/components/MultiAgentAccelerator/Sidebar/Sidebar.tsx:1]: Lots of diffs here, but mostly _removing_ code related to setting, testing, and saving the Neuro SAN URL, which is all now in the Settings dialog. +180, -322 -- not bad.

File: packages/ui-common/components/MultiAgentAccelerator/Sidebar/Sidebar.tsx [hunk 21]
```diff
@@ -14,40 +14,28 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 import AddBoxRounded from "@mui/icons-material/AddBoxRounded"
-import CheckCircleOutlinedIcon from "@mui/icons-material/CheckCircleOutlined"
```
Comment (dsargent) [inline packages/ui-common/components/MultiAgentAccelerator/Sidebar/Sidebar.tsx:17]: Lots of code removed relating to the URL popup.

File: packages/ui-common/components/MultiAgentAccelerator/Sidebar/Sidebar.tsx [hunk 22]
```diff
@@ -337,209 +285,119 @@ export const Sidebar: FC<SidebarProps> = ({
...
-                    items={treeViewItems}
-                    multiSelect={false}
-                    onExpandedItemsChange={(_event, itemIds) => setExpandedItems(itemIds)}
-                    onSelectedItemsChange={handleSelectedItemsChange}
-                    selectedItems={selectedItem}
-                    slotProps={{
-                        item: {onDeleteNetwork, onEditNetwork: handleEditNetworkWithSelect} as AgentNetworkNodeProps,
-                    }}
-                    slots={{item: AgentNetworkTreeItem as RichTreeViewSlots["item"]}}
+    const toStatusColor = () => {
```
Comment (dsargent) [inline packages/ui-common/components/MultiAgentAccelerator/Sidebar/Sidebar.tsx:288]: Remove all the URL dialog stuff and instead, add this status light with the handy tooltip.

File: packages/ui-common/components/MultiAgentAccelerator/Sidebar/Sidebar.tsx [hunk 23]
```diff
@@ -337,209 +285,119 @@ export const Sidebar: FC<SidebarProps> = ({
...
+            selectedItems={selectedItem}
+            slotProps={{
+                item: {onDeleteNetwork, onEditNetwork: handleEditNetworkWithSelect} as AgentNetworkNodeProps,
+            }}
+            slots={{item: AgentNetworkTreeItem as RichTreeViewSlots["item"]}}
+        />
+    )
+
+    return (
+        <SidebarAside id={`${id}-sidebar`}>
```
Comment (dsargent) [inline packages/ui-common/components/MultiAgentAccelerator/Sidebar/Sidebar.tsx:398]: Did my usual refactoring of rendering logic into functions, while keeping the mainline lean and mean.

File: packages/ui-common/__tests__/components/MultiAgentAccelerator/MultiAgentAccelerator.test.tsx [hunk 24]
```diff
@@ -216,7 +216,7 @@ const renderMultiAgentAcceleratorPage = () =>
     render(
         <SnackbarProvider>
             <MultiAgentAccelerator
-                backendNeuroSanApiUrl={NEURO_SAN_SERVER_URL}
+                defaultNeuroSanUrl={NEURO_SAN_SERVER_URL}
```
Comment (dsargent) [inline packages/ui-common/__tests__/components/MultiAgentAccelerator/MultiAgentAccelerator.test.tsx:219]: Renamed field

File: packages/ui-common/__tests__/components/MultiAgentAccelerator/MultiAgentAccelerator.test.tsx [hunk 25]
```diff
@@ -1343,6 +1343,10 @@ describe("MultiAgentAccelerator", () => {
 
             // Click through remaining steps and verify their content shows up
             for (const step of MAIN_TOUR_STEPS.slice(1)) {
+                // Make sure step at least appears in the DOM, though due to mocks we may get false negatives here
```
Comment (dsargent) [inline packages/ui-common/__tests__/components/MultiAgentAccelerator/MultiAgentAccelerator.test.tsx:1346]: Tighten this test up a bit, but there are still Swiss-cheese gaps in it.

For one, we mock a lot of things, so the tour still _appears_ to work even if we've pulled the rug and changed the UI behind its back. So IDs in the implementation can drift, and this test will still blithely pass. We don't have a single point of truth for the IDs. Even if we did, as long as we're mocking things like `AgentFlow`, we can't be sure the Tour will actually work in the wild, since `AgentFlow` could somehow have been broken and stop rendering what we think it renders, even if we use the same IDs.

So a few items could be improved:

1) Single source of truth for IDs -or-
2) Have a test that doesn't mock everything, a sort of "Tour-only" test. Could get unwieldy though.
3) Pass refs to the Tour (maybe a dictionary of steps to refs or something), rather than having blind faith in IDs. At least then tsc will fail if we remove a ref that the Tour depends on.

All of the above is extra work, beyond the scope of this PR.

File: packages/ui-common/__tests__/components/MultiAgentAccelerator/Sidebar/Sidebar.test.tsx
Comment (dsargent) [inline packages/ui-common/__tests__/components/MultiAgentAccelerator/Sidebar/Sidebar.test.tsx:1]: A lot of the tests in here are now moot, since I moved the URL config to the Settings dialog (which now has its own tests around that). So a lot of machete-ing here.

File: packages/ui-common/__tests__/components/MultiAgentAccelerator/Sidebar/Sidebar.test.tsx [hunk 26]
```diff
@@ -163,14 +129,22 @@ describe("SideBar", () => {
         mockSelectedTreeItemId = undefined
 
         user = userEvent.setup()
-        ;(testConnection as jest.Mock).mockResolvedValue({success: true, status: "ok", version: TEST_VERSION})
+        const testConnectionMock = jest.mocked(testConnection)
```
Comment (dsargent) [inline packages/ui-common/__tests__/components/MultiAgentAccelerator/Sidebar/Sidebar.test.tsx:132]: My new favorite style for safe, strongly-typed mocks that scream if you make typos or mess up properties.

File: packages/ui-common/__tests__/components/MultiAgentAccelerator/Sidebar/Sidebar.test.tsx [hunk 27]
```diff
@@ -163,14 +129,22 @@ describe("SideBar", () => {
...
+            status: "ok",
+            success: true,
+            version: TEST_VERSION,
+        } satisfies TestConnectionResult)
 
         // Reset settings store
         useSettingsStore.getState().resetSettings()
     })
 
-    it.each(["light", "dark"] as PaletteMode[])("should render correctly with darkMode=%s", async (mode) => {
```
Comment (dsargent) [inline packages/ui-common/__tests__/components/MultiAgentAccelerator/Sidebar/Sidebar.test.tsx:172]: `Sidebar` no longer does theme hacks, so this test that pivots on the theme doesn't need to do that any more.

File: packages/ui-common/__tests__/components/MultiAgentAccelerator/Sidebar/Sidebar.test.tsx [hunk 28]
```diff
@@ -179,25 +153,117 @@ describe("SideBar", () => {
...
         // Clicking on a network should call the setSelectedNetwork function
         const network = await screen.findByText(TEST_AGENT_MATH_GUY_DISPLAY)
         await user.click(network)
 
         // setSelectedNetwork should be called
         expect(setSelectedNetwork).toHaveBeenCalledTimes(1)
         expect(setSelectedNetwork).toHaveBeenCalledWith(`${TEST_AGENTS_FOLDER}/${TEST_AGENT_MATH_GUY}`)
+    })
+
+    it("should show correct Neuro SAN status", async () => {
```
Comment (dsargent) [inline packages/ui-common/__tests__/components/MultiAgentAccelerator/Sidebar/Sidebar.test.tsx:165]: New tests for status light

File: packages/ui-common/__tests__/components/MultiAgentAccelerator/Sidebar/Sidebar.test.tsx [hunk 29]
```diff
@@ -179,25 +153,117 @@ describe("SideBar", () => {
...
+        const unknownStatusTooltip = await screen.findByRole("tooltip", {name: "Neuro-san server status"})
+        within(unknownStatusTooltip).getByText(statusMessage)
+        within(unknownStatusTooltip).getByText("offline")
+        within(unknownStatusTooltip).getByText("unknown")
+        within(unknownStatusTooltip).getByText(new RegExp(`${String(httpStatus.IM_A_TEAPOT + 1)}.*Unknown status`, "u"))
+    })
 
-        // Mousing over the cog should show the tooltip with the version and URL
-        const settingsButton = await screen.findByRole("button", {name: /Agent Network Settings/u})
+    it.each([
```
Comment (dsargent) [inline packages/ui-common/__tests__/components/MultiAgentAccelerator/Sidebar/Sidebar.test.tsx:244]: Just to hit coverage numbers.

File: packages/ui-common/__tests__/components/MultiAgentAccelerator/Sidebar/Sidebar.test.tsx [hunk 30]
```diff
@@ -714,7 +560,7 @@ describe("SideBar", () => {
         // runOnlyPendingTimers fires only timers already in the queue, so the
         // 5000ms sparkle-remove timer registered inside the callback won't fire here.
         act(() => {
-            jest.runOnlyPendingTimers()
+            jest.advanceTimersByTime(50)
```
Comment (dsargent) [inline packages/ui-common/__tests__/components/MultiAgentAccelerator/Sidebar/Sidebar.test.tsx:563]: This was now running afoul of the _new_ interval timer that periodically polls Neuro SAN to see if it's online. Since the previous logic was running _all_ timers, that was also causing the Neuro SAN interval to elapse and the ping to run, causing state updates and re-renders. By limiting the synthetic interval to 50ms, we ensure only the sparkle timer runs.

File: packages/ui-common/__tests__/components/MultiAgentAccelerator/Sidebar/Sidebar.test.tsx [hunk 31]
```diff
@@ -744,7 +590,7 @@ describe("SideBar", () => {
 
         // Fire the 50ms timer — selectedNode is null, so the callback is a no-op.
         act(() => {
-            jest.runOnlyPendingTimers()
+            jest.advanceTimersByTime(50)
```
Comment (dsargent) [inline packages/ui-common/__tests__/components/MultiAgentAccelerator/Sidebar/Sidebar.test.tsx:593]: Ditto on timer issues.

File: packages/ui-common/__tests__/components/Settings/SettingsDialog.test.tsx [hunk 32]
```diff
@@ -98,6 +102,173 @@ describe("SettingsDialog", () => {
         expect(onCloseMock).toHaveBeenCalledTimes(1)
     })
 
+    describe("External services", () => {
```
Comment (dsargent) [inline packages/ui-common/__tests__/components/Settings/SettingsDialog.test.tsx:105]: Tests for the new section for entering the Neuro SAN URL.

File: packages/ui-common/__tests__/components/Settings/SettingsDialog.test.tsx [hunk 33]
```diff
@@ -119,12 +290,12 @@ describe("SettingsDialog", () => {
             await user.type(inputBox, testApiKey)
 
             // Click "Test" button to (pretend) "test" the key
-            const testButton = within(apiKeyInput).getByRole("button", {name: /Test/u})
+            const testButton = within(apiKeyInput).getByRole("button", {name: "Test"})
```
Comment (dsargent) [inline packages/ui-common/__tests__/components/Settings/SettingsDialog.test.tsx:293]: Changed some of these regex queries to plain old strings. I think it's faster and easier to read when we know the exact value we're looking for, and in fact better expresses our intent.

File: packages/ui-common/components/Settings/SettingsDialog.tsx [hunk 34]
```diff
@@ -320,6 +434,97 @@ export const SettingsDialog: FC<SettingsDialogProps> = ({id, isOpen, logoService
...
+                        disabled={neuroSanUrlSaveDisabled}
+                        onClick={handleSaveNeuroSanUrl}
+                        size="small"
+                        variant="contained"
+                    >
+                        Save
+                    </Button>
+                    <Button
+                        disabled={
+                            neuroSanUrlInput === stripProtocol(defaultNeuroSanUrl) &&
```
Comment (swensel) [inline packages/ui-common/components/Settings/SettingsDialog.tsx]: Maybe move this disabled check logic to its own variable and refer to that here.

**Review by swensel — DISMISSED** [review]
LGTM. Great feature! 🎉

File: packages/ui-common/components/Settings/SettingsDialog.tsx (hunk 34 above)
Comment (dsargent) [reply packages/ui-common/components/Settings/SettingsDialog.tsx]: Agreed. I did that in other cases but this was a "post-Devin suggestion". Will do it.

**Review by swensel — APPROVED** [review]
Thanks for making changes.
